moquette/broker/src/main/java/io/moquette/broker/Session.java

```diff
@@ -113,6 +113,34 @@ public int compareTo(Delayed o) {
         this.sessionQueue = sessionQueue;
     }
 
+    /**
+     * This method terminates a Session, which involves disconnecting the client,
+     * if one is connected, and releasing all references to unacknowledged publishes.
+     */
+    void terminateSession() {
+        MQTTConnection connection = this.mqttConnection;
+        if (connection != null) {
+            connection.dropConnection();
+        }
+        disconnect();
+
+        // Release queued messages
```

Comment:
can you explain why you want to release queued messages

Reply:
Will update